### PR TITLE
detect and convert date(time) fields for target project's validation

### LINKED ISSUE #16: Convert dates to the format of the target field
Convert dates to the format of the target field before attempt to write them. As REDCap stores dates as YMD, they are fetch as YMD. Yet the data entry rules of a non-YMD fields will chew and/or reject the date.

### PR DESCRIPTION
**Pull request by ChemiKyle**
Close #16 

Conversion is handled on the backend, so values will appear adjusted in the confirmation modal dialog as well.

### CODE REVIEW

File: ExternalModule.php [hunk 1]
```diff
@@ -100,6 +100,49 @@ function ( $value, $source_key ) use ( $mapping, &$source_person_data, $all_pers
         return $target_person_data;
     }
 
+    // Copied nearly exactly from the DataQuality class because it's a private function
+    // TODO: utilize DateTimeRC::datetimeConvert, but this does all the lifting
+    private function convertDateFormat($field, $value) {
```
Comment (mbentz-uf) [inline ExternalModule.php:105]: Adding a comment specifying the target date format (ymd) could be helpful

**Review by mbentz-uf — APPROVED** [review]
All date formats properly convert to ymd.

**Review by mbentz-uf — APPROVED** [review]

**Comment by ChemiKyle** [comment]
Your comments are a bit confusing, REDCap stores all date(time) fields as ymd(hms) on the backend, the goal of this PR is to convert to the target field's validation type.

This also leads me to realize there are no test project XML files to set this up for testing! Normally PRs have a testing section, but since this was originally intended for Philip I was lazy.

To test:
1. Set up Source and Target projects using [APF's example project XML](https://github.com/ctsit/auto_populate_fields/)
1. Enable this module on the target project, and map the fields in the "Test dates" form
1. Fill in a handful of records in the Source project
1. Return to the Target project, create a record, and attempt to use SAPDAP to clone any of the record(s) you created in the Source project

File: ExternalModule.php (hunk 1 above)
Comment (mbentz-uf) [reply ExternalModule.php:105]: Sorry I misunderstood, the conversion _from_ ymd to the field format works properly.

**Pull request merged**